<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find walker_planner)/config/rosconsole.config" />

    <arg name="debug" default="false"/>

    <!-- <arg name="planner_id" default="mhastar.bfs.bfs_fullbody.manip"/> -->
    <arg name="planner_id" default="mhastar.bfs.bfs_fullbody.euclid_fullbody.manip"/>    
    <arg name="epsilon" default="50.0"/>    
    <arg name="plan_time" default="300.0"/>

    <arg name="scenario" default="multi_room"/>
    <arg name="exp" default="0"/>

    <arg name="post_process" default="false"/>
    <arg name="animate" default="true"/>
    <arg name="logging" default="false"/>
    <arg name="log_filename" default="$(find walker_planner)/critical_results/$(arg scenario)/test.csv"/>    

    <arg name="rvizconfig" default ="$(find walker_planner)/config/default_walker_cfg.rviz" />

    <include file="$(find walker_planner)/launch/planning_context_walker.launch"/>

    <arg unless="$(arg debug)" name="launch-prefix" value=""/>
    <arg     if="$(arg debug)" name="launch-prefix" value="gdb -ex run --args"/>
    <env name="SMPL_VISUALIZE_CONFIG_FILE" value="$(find walker_planner)/config/smpl_visual.ini"/>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find walker_planner)/config/rosconsole.config"/>        

    <!-- tf -->
    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/> -->
    <!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" /> -->

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

    <node pkg="walker_planner" type="test_miastar_planner" name="test_miastar_planner"
            launch-prefix="$(arg launch-prefix)" output="screen">
        <param name="planning_frame" value="dummy_base"/>

        <!-- robot model (for planning) -->
        <rosparam param="robot_model">
            group_name: full_body
            planning_joints:
                x
                y
                theta
                right_j1
                right_j2
                right_j3
                right_j4
                right_j5
                right_j6
                right_j7
            kinematics_frame:
                planar_1
            chain_tip_link:
                right_palm_dummy_link
        </rosparam>

        <!-- collision checking -->
        <rosparam command="load" file="$(find walker_planner)/config/collision_model_walker.yaml"/>

        <!-- planner params -->
        <rosparam command="load" file="$(find walker_planner)/config/walker_right_arm.yaml"/>        
        <!-- <param name="planning/mprim_filename" value="$(find walker_planner)/config/walker_multi_rep.mprim"/> -->
        <param name="planning/mprim_filename" value="$(find walker_planner)/config/walker.mprim"/>        
        <param name="allowed_planning_time" value="300.0"/>

        <param name="planner_id" value="$(arg planner_id)"/>
        <param name="epsilon" value="$(arg epsilon)"/> 
        <param name="allowed_planning_time" value="$(arg plan_time)"/>        
        <param name="scenario" type="string" value="$(arg scenario)"/>        
        <param name="exp_id" type="string" value="$(arg exp)"/>


        <param name="use_post_processing" value="$(arg post_process)"/>                
        <param name="use_visualization" value="$(arg animate)"/>
        <param name="use_logging" value="$(arg logging)"/>
        <param name="log_filename" value="$(arg log_filename)"/>        

        <!-- experiment params -->
        <param name="logging" value="$(arg logging)"/>        
        <param name="exp_id" type="string" value="$(arg exp)"/>                        
        <rosparam command="load" file="$(find
        walker_planner)/critical_experiments/$(arg scenario)/$(arg scenario)$(arg exp).yaml"/>

        <!-- environment params -->
        <rosparam command="load" file="$(find walker_planner)/env/multi_room_map.yaml" />
        <!-- <param name="object_filename" value="$(find walker_planner)/env/multi_rooms.env"/> -->

    </node>
</launch>
